backend/core/src/main/java/org/sonarsource/sonarlint/core/ConnectionService.java

```
@@ -232,4 +237,13 @@ public Map<String, String> getProjectNamesByKey(Either<TransientSonarQubeConnect
     return projectNamesByKey;
   }
 
+  private boolean isAnonymousConnection(Either<TransientSonarQubeConnectionDto, TransientSonarCloudConnectionDto> transientConnection) {
+    Either<TokenDto, UsernamePasswordDto> credentials = transientConnection
+      .map(TransientSonarQubeConnectionDto::getCredentials, TransientSonarCloudConnectionDto::getCredentials);
+
+    boolean isTokenEmpty = credentials.getLeft() == null || credentials.getLeft().getToken() == null || credentials.getLeft().getToken().isEmpty();
+    boolean isUserPassEmpty = credentials.getRight() == null || credentials.getRight().getUsername() == null || credentials.getRight().getPassword() == null;
+
+    return isTokenEmpty && isUserPassEmpty;
+  }
```

These fields are not nullable, the client should likely throw an error instead of sending null values 🤔

---

Hmm, on the `ConnectionAwareHttpClientProvider.java` there were already some null checks for those.

When a user imports the VSCode settings to Cursor, the secure storage is not imported so the tokens are null. However, the null token is reported as valid from the `serverApi.authentication().validate`, like an anonymous user and because we didn't have the `null` checks here (to prevent anonymous connections), the UI on Cursor connections list displays them as `ok`. All the other connected mode actions that use the `ConnectionAwareHttpClientProvider` are failing because of invalid credentials. 🤔